enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,391 @@
+---
+title: downstream-operator-sdk
+authors:
+  - "@jmrodri"
+reviewers:
+  - "@joelanford"
+  - "@asmacdo"
+  - "@fabianvf"
+  - "@camilamacedo86"
+  - "@shawn-hurley"
+  - "@gallettilance"
+  - "@kevinrizza"
+approvers:
+  - "@joelanford"
+  - "@asmacdo"
+  - "@fabianvf"
+creation-date: 2020-09-10
+last-updated: 2020-09-10
+status: implementable
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Downstream Operator SDK
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Operator SDK currently ships the Ansible and Helm Operator base images
+downstream. The purpose of this enhancement is to add a supported Operator SDK
+base image as well as an operator-sdk binary. In addition, to add more
+automation to syncing upstream with downstream.
+
+## Motivation
+
+Deliver a properly supported Operator SDK downstream to support OpenShift
+developers.
+
+### Goals
+
+The primary goal is to deliver a new base image with the Operator SDK that
+can be used by operator developers. Deliver an operator-sdk binary similar to
+how oc is delivered to customers. Continue to deliver the Ansible and Helm
+operator base image.
+
+Another goal is to automate the syncing of upstream to downstream to make it
+easier to bring in upstream releases.
+
```

Comment:
In sdk we setup the env to run the tests against Kind. Maybe an end goal would have the sdk/upstream tests set up to be executed against OCP.  PS. I am not sure if they will work without any adjustment. It might require some effort. See that we have if is KIND load the image : https://github.com/operator-framework/operator-sdk/blob/master/test/e2e-helm/e2e_helm_suite_test.go#L117-L121. So, we might need to do spend an effort here.

Reply:
We run the downstream tests against OCP now. All the tests in `ci/tests` run in OCP. So there will be no kind cluster needed downstream. And we do not want to run upstream against OCP.

Reply:
However, the ci/test that has there now are outdated and I do not think that they make sense any more. I mean, they are for 0.9.x and they will not work for 1.0+. So, IHMO we would need to make the upstream tests know if is kind or OCP and do what is specific. See;  https://github.com/operator-framework/operator-sdk/blob/master/test/e2e-helm/e2e_helm_suite_test.go#L93

Reply:
We spoke about that all fine. 
We checked that the upstream tests would work and we just need to figure out how to build/push the images to prow.

Reply:
we can close this one.